Example/IGListKitExamples/ViewControllers/MixedDataViewController.swift

```diff
@@ -71,13 +71,8 @@ class MixedDataViewController: UIViewController, IGListAdapterDataSource {
     //MARK: IGListAdapterDataSource
 
     func objects(for listAdapter: IGListAdapter) -> [IGListDiffable] {
-        var items = [IGListDiffable]()
-        for obj in data {
-            if selectedClass == nil || type(of: obj) == selectedClass! {
-                items.append(obj as! IGListDiffable)
-            }
-        }
-        return items
+        guard selectedClass != nil else { return data.map { $0 as! IGListDiffable } }
+        return data.filter { type(of: $0) == selectedClass! }.map { $0 as! IGListDiffable }
```

Comment:
actually, now i remember the issues I saw with this. scratch that first comment. 😄 

the previous version combined the 2 conditions in a single `if`. Now this is broken up and harder to follow, not to mention we've written `.map { $0 as! IGListDiffable }` twice. 😊 

why can't the initial filter contain both conditions? couldn't this be something like this? 

```swift
return data.filer { 
        if let class = selectedClass { 
            return type(of: $0) == class 
        } 
        return false
        }.map { $0 as! IGListDiffable }
```

I think this is more readable + more concise.